port/blueprint-permissions/schema.go

```diff
@@ -0,0 +1,232 @@
+package blueprint_permissions
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+
+	"github.com/hashicorp/terraform-plugin-framework-validators/mapvalidator"
+	"github.com/hashicorp/terraform-plugin-framework-validators/stringvalidator"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/booldefault"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+func getAssigneeProps(permName string) map[string]schema.Attribute {
+	return map[string]schema.Attribute{
+		"users": schema.ListAttribute{
+			MarkdownDescription: fmt.Sprintf("Users with %+v permissions", permName),
+			Optional:            true,
+			ElementType:         types.StringType,
+		},
+		"roles": schema.ListAttribute{
+			MarkdownDescription: fmt.Sprintf("Roles with %+v permissions", permName),
+			Optional:            true,
+			ElementType:         types.StringType,
+		},
+		"teams": schema.ListAttribute{
+			MarkdownDescription: fmt.Sprintf("Teams with %+v permissions", permName),
+			Optional:            true,
+			ElementType:         types.StringType,
+		},
+		"owned_by_team": schema.BoolAttribute{
+			MarkdownDescription: "Owned by team",
+			Optional:            true,
+			Computed:            true,
+			Default:             booldefault.StaticBool(false),
+		},
+	}
+
+}
+
+func BlueprintPermissionsSchema() map[string]schema.Attribute {
+	return map[string]schema.Attribute{
+		"id": schema.StringAttribute{
+			Computed: true,
+		},
+		"blueprint_identifier": schema.StringAttribute{
+			Required: true,
+		},
+		"entities": schema.SingleNestedAttribute{
+			MarkdownDescription: "Entities permissions to read the blueprint",
+			Required:            true,
+			Attributes: map[string]schema.Attribute{
+				"register": schema.SingleNestedAttribute{
+					Required:   true,
+					Attributes: getAssigneeProps("register"),
+				},
+				"unregister": schema.SingleNestedAttribute{
+					Required:   true,
+					Attributes: getAssigneeProps("unregister"),
+				},
+				"update": schema.SingleNestedAttribute{
+					Required:   true,
+					Attributes: getAssigneeProps("update"),
+				},
+				"update_metadata_properties": schema.SingleNestedAttribute{
+					Required: true,
```

Comment:
missing description, also this is a "special" property we added which isn't aligned with the API, so we should add more explanation about it

Reply:
Added